diff --git a/dist/index.js b/dist/index.js
index 24a7fe151b255778b2fb83bea94c5d230dfbfd18..89f66bc766f18adcf4316d1f0cca63f0c265bbe5 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -3,12 +3,14 @@ import { basename, join, dirname } from "node:path";
 import * as esbuild from "esbuild";
 import { mkdtemp, writeFile } from "node:fs/promises";
 import { tmpdir } from "node:os";
-const DEFAULT_INCLUDE_REGEX = /\.gen\.[mc]?[jt]sx?(?:\?.+?)?$/;
+import { pathToFileURL } from "node:url";
+const DEFAULT_INCLUDE_REGEX = /\.gen\.[mc]?[jt]sx?(?:[?&].+?)?$/;
 function stripQueryArgs(path) {
-    return path.replace(/\?.*$/, "");
+    return path.replace(/[?&][^.]+$/, "");
 }
 const SYM_TRANSPILE = Symbol();
 const PLUGIN_NAME = "generate";
+const SYM_PATH_KEY = Symbol();
 export function generate({ exclude, include = DEFAULT_INCLUDE_REGEX, esbuildOptions: _esbuildOptions = {}, emitDts = true, } = {}) {
     const matcher = createFilter(include, exclude);
     function getBuildOptions(id) {
@@ -26,9 +28,10 @@ export function generate({ exclude, include = DEFAULT_INCLUDE_REGEX, esbuildOpti
         const dir = dirname(id);
         const outfile = join(tempDir, "file.cjs");
         const providedOptions = await getBuildOptions(id);
+        const normalizedId = stripQueryArgs(id);
         const result = await esbuild.build({
             ...providedOptions,
-            entryPoints: [id],
+            entryPoints: [normalizedId],
             outfile,
             bundle: true,
             minify: false,
@@ -52,7 +55,8 @@ export function generate({ exclude, include = DEFAULT_INCLUDE_REGEX, esbuildOpti
                 this.addWatchFile(path);
             }
         }
-        let mod = (await import(outfile));
+        let requireURL = process.platform === "win32" ? pathToFileURL(outfile).href : outfile;
+        let mod = (await import(requireURL));
         // try to unwrap cjs default export
         if (Object.keys(mod).length === 1 && "default" in mod) {
             mod = mod.default;
@@ -110,25 +114,45 @@ export function generate({ exclude, include = DEFAULT_INCLUDE_REGEX, esbuildOpti
             return;
         }
         else {
-            const outputDtsWithThing = id.replace(/\.[mc]?[jt]sx?$/, "?gen.d.ts");
+            const outputDtsWithThing = id.replace(/\.[mc]?[jt]sx?$/, "&gen.d.ts");
             await writeFile(outputDtsWithThing, dts);
         }
     }
     return {
         name: "generate",
+        async resolveId(source, importer, options) {
+            const strippedSource = stripQueryArgs(source)
+            const stripped = (await this.resolve(strippedSource, importer, options))?.id ?? "";
+            let resolvedSource = source;
+            if (stripped.match(/\.[^?/\\.&]+$/)) {
+                resolvedSource = stripped + source.slice(strippedSource.length);
+            }
+            if (!matcher(resolvedSource)) {
+                return null;
+            }
+            return {
+                id: stripped,
+                meta: {
+                    [PLUGIN_NAME]: SYM_TRANSPILE,
+                    [SYM_PATH_KEY]: resolvedSource,
+                }
+            }
+        },
         async load(id) {
-            if (!matcher(id)) {
+            const meta = this.getModuleInfo(id)?.meta;
+            if (meta?.[PLUGIN_NAME] !== SYM_TRANSPILE) {
                 return null;
             }
-            const result = await transform.call(this, id);
+            const resolvedId = meta[SYM_PATH_KEY];
+            if (!resolvedId) {
+                throw new Error("missing resolved id");
+            }
+            const result = await transform.call(this, resolvedId);
             if (emitDts) {
-                await generateAndWriteDts.call(this, stripQueryArgs(id), result);
+                await generateAndWriteDts.call(this, stripQueryArgs(resolvedId), result);
             }
             return {
                 code: result,
-                meta: {
-                    [PLUGIN_NAME]: SYM_TRANSPILE,
-                },
             };
         },
     };
