diff --git a/dist/dtsBundler.js b/dist/dtsBundler.js
index 4a74f7b1e8856064288314800eb891a5520959bc..106b071baf66fa865d40cbe35c7d0794704a4e87 100644
--- a/dist/dtsBundler.js
+++ b/dist/dtsBundler.js
@@ -58,7 +58,7 @@ async function generateDtsViaRollup(ctx, { entry, files, aliases }) {
 export async function generateAndWriteDts(id, code, virtualFiles, banner = "") {
     try {
         let generatedFs = await generateDtsNoBundle(this, id, code, virtualFiles);
-        const mainDtsPath = pathToDtsPath(id);
+        const mainDtsPath = pathToDtsPath(normalize(id));
         let result = "";
         if (!virtualFiles.hasGeneratedFiles(id)) {
             result = generatedFs.get(mainDtsPath) ?? "";
