diff --git a/client.d.ts b/client.d.ts
index 3a9dd88f771f0d88310d8441543af477269e3973..792d3390a1be9b87127116721f9efafa1607efc2 100644
--- a/client.d.ts
+++ b/client.d.ts
@@ -5,35 +5,35 @@ type CSSModuleClasses = { readonly [key: string]: string }
 
 declare module '*.module.css' {
   const classes: CSSModuleClasses
-  export default classes
+  export = classes
 }
 declare module '*.module.scss' {
   const classes: CSSModuleClasses
-  export default classes
+  export = classes
 }
 declare module '*.module.sass' {
   const classes: CSSModuleClasses
-  export default classes
+  export = classes
 }
 declare module '*.module.less' {
   const classes: CSSModuleClasses
-  export default classes
+  export = classes
 }
 declare module '*.module.styl' {
   const classes: CSSModuleClasses
-  export default classes
+  export = classes
 }
 declare module '*.module.stylus' {
   const classes: CSSModuleClasses
-  export default classes
+  export = classes
 }
 declare module '*.module.pcss' {
   const classes: CSSModuleClasses
-  export default classes
+  export = classes
 }
 declare module '*.module.sss' {
   const classes: CSSModuleClasses
-  export default classes
+  export = classes
 }
 
 // CSS
